### PR TITLE
Kql validations For Hunting Queries 

### PR DESCRIPTION
**Pull request by v-vdixit**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: .script/tests/KqlvalidationsTests/CustomTables/Ubiquiti.json [hunk 1]
```diff
@@ -145,9 +145,13 @@
...
-        {
-            "Name": "Message",
-            "Type": "String"
-        }
+      {
+        "Name": "Message",
+        "Type": "String"
+      },
+      {
+        "Name": "DnsQueryName",
```
Comment (oshezaf) [inline .script/tests/KqlvalidationsTests/CustomTables/Ubiquiti.json:153]: I could not find the field DnsQueryName in the [parser](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Ubiquiti%20UniFi/Parsers/UbiquitiAuditEvent.txt)

**Review by oshezaf — CHANGES_REQUESTED** [review]
I have found a few issues in a cursory review of the PR. I suspect this a wider challenge with the PR.

**Comment by ashishsyal** [comment]
Please hold the merge of the PR - until Mahesh reviews.

**Comment by v-rucdu** [comment]
@v-vdixit is working on fixing the review comments as well as the validation fixes... Thanks!!!

File: .script/tests/KqlvalidationsTests/CustomTables/Ubiquiti.json (hunk 1 above)
Comment (v-vdixit) [reply .script/tests/KqlvalidationsTests/CustomTables/Ubiquiti.json]: Yes, @oshezaf I checked and removed it and made other changes too in accordance with the respective parsers, Thanks!

File: .script/tests/KqlvalidationsTests/CustomTables/imRegistry.json [hunk 2]
```diff
@@ -44,6 +44,13 @@
         {"Name":"ParentProcessGuid","Type":"string"},
         {"Name":"Dvc","Type":"string"},
         {"Name":"User","Type":"string"},
-        {"Name":"Process","Type":"string"}
+      {
+        "Name": "Process",
+        "Type": "string"
+      },
+      {
+        "Name": "Username",
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply .script/tests/KqlvalidationsTests/CustomTables/imRegistry.json:52]: Hi @oshezaf I checked and confirmed from the parser for Username and I found it added here https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoft365D.yaml, that is why I added it to imRegistry. Thanks!

[the remark this answers is not in the document]
Comment (oshezaf) [reply .script/tests/KqlvalidationsTests/CustomTables/imRegistry.json:52]: It is a bug in this parser that needs to be fixed. The correct field name as document [here](https://learn.microsoft.com/en-us/azure/sentinel/registry-event-normalization-schema) is User

File: .script/tests/KqlvalidationsTests/CustomTables/CiscoSEGEvent.json [hunk 3]
```diff
@@ -577,9 +577,9 @@
             "Name": "NetworkDirection",
             "Type": "String"
         },
-        {
-            "Name": "AdditionalFields",
-            "Type": "Object"
-        }
+      {
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/CiscoSEGEvent.json:580]: Do not see any difference between old code and new code.

File: .script/tests/KqlvalidationsTests/CustomTables/Cloudflare.json [hunk 4]
```diff
@@ -353,9 +353,13 @@
             "Name": "MatchIndex",
             "Type": "Double"
         },
-        {
-            "Name": "OriginatorRayID",
-            "Type": "String"
-        }
+      {
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/Cloudflare.json:356]: Do not see any difference between old code and new code.

File: .script/tests/KqlvalidationsTests/CustomTables/CyberSixgill_Alerts_CL.json [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  {
+    "Name": "Computer",
+    "Type": "string"
+  },
+  {
+    "Name": "RawData",
+    "Type": "string"
+  },
+  {
+    "Name": "alert_name_s",
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/CyberSixgill_Alerts_CL.json:33]: why schema having column names like columns in the log analytics table.

File: .script/tests/KqlvalidationsTests/CustomTables/GitHubRepo.json [hunk 6]
```diff
@@ -40,8 +40,8 @@
     {
       "Name": "Summary",
       "Type": "String"
-    },	
-	{
+    },
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/GitHubRepo.json:43]: Looks like there is no change in the file.

File: .script/tests/KqlvalidationsTests/CustomTables/GitHubRepo.json [hunk 7]
```diff
@@ -81,7 +81,7 @@
       "Name": "PushPermissions",
       "Type": "String"
     },
-	{
+    {
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/GitHubRepo.json:84]: Looks like there is no change in the file.

File: .script/tests/KqlvalidationsTests/CustomTables/Okta_CL.json [hunk 8]
```diff
@@ -40,6 +40,10 @@
     {
       "Name": "actor_alternateId_s",
       "Type": "String"
+    },
+    {
+      "Name": "target_s",
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/Okta_CL.json:45]: why schema having column names like columns in the log analytics table. Custom table should have column with final field names.

File: .script/tests/KqlvalidationsTests/CustomTables/SenservaPro_CL.json [hunk 9]
```diff
@@ -60,6 +60,10 @@
     {
       "Name": "Group_s",
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/SenservaPro_CL.json:61]: Custom table schema having column names like column_x. Custom table should have column with final field names.

File: .script/tests/KqlvalidationsTests/CustomTables/ZoomLogs.json [hunk 10]
```diff
@@ -40,7 +40,26 @@
     {
       "Name": "payload_object_participant_user_name_s",
       "Type": "String"
+    },
+    {
+      "Name": "RoomEvents",
+      "Type": "String"
+    },
+    {
+      "Name": "payload_object_room_name_s",
```
Comment (mkchiliveri) [inline .script/tests/KqlvalidationsTests/CustomTables/ZoomLogs.json:49]: Custom table schema having column names like column_x. Custom table should have column with final field names.

File: ASIM/lib/functions/ASIM_Enrich_IdentityInfo.yaml [hunk 11]
```diff
@@ -75,8 +75,7 @@ FunctionQuery: |
             UsernameType = "UPN",
             UserIdType = "AadId",
             UserType = iff(UserType == "Guest", UserType, ""),
-            UserUpn = Username
-    ;
+            UserUpn = Username;
```
Comment (mkchiliveri) [inline ASIM/lib/functions/ASIM_Enrich_IdentityInfo.yaml:78]: Looks like there is no change in the file.

File: Hunting Queries/GitHub/Unusual Number of Repository Clones.yaml [hunk 12]
```diff
@@ -10,9 +10,9 @@ relevantTechniques:
 query: |
 
   let min_t = toscalar(GitHubRepo
-  | summarize min(timestamp_t));
```
Comment (mkchiliveri) [inline Hunting Queries/GitHub/Unusual Number of Repository Clones.yaml:13]: Check this again, as i feel there may be timestamp field available in the data.

**Comment by v-vdixit** [comment]
Following up on the related changes in the following PRs #7279 #7280, #7282, #7285, #7288  #7295 #7301

**Pull request closed without merging**